File: 2000 98 & ME/QuickStart/howto/samples/Services/Timers/scheduler/VB/Scheduler.vb
Imports System
Imports System.Diagnostics
Imports System.Threading
Imports System.Timers
Imports System.Timers.Design
Imports System.WinForms

Namespace Scheduler

    Module Scheduler

        Public executablePath As String

        Public Sub Main()
            Dim args As String()
            Dim sampleName As String
            args = Environment.GetCommandLineArgs()
            sampleName = args(0)

            If (args.Length <> 2) Then
                Console.WriteLine("Usage: " + sampleName + " <executable>")
                Exit Sub
            End If

            executablePath = args(1)

            Dim aSchedule As Schedule
            aSchedule = New Schedule

            Dim dlg As RecurrencePatternsDialog
            dlg = New RecurrencePatternsDialog

            dlg.RecurrencePatterns = aSchedule.RecurrencePatterns
            dlg.ShowDialog()

            aSchedule.RecurrencePatterns.All = dlg.RecurrencePatterns.All
            AddHandler aSchedule.EventOccurred, AddressOf OnScheduledEvent
            aSchedule.Enabled = True

            Console.WriteLine("Press 'q' to quit the sample")
            While (Console.Read() <> 113)
            End While
        End Sub

        Public Sub OnScheduledEvent(ByVal source As Object, ByVal e As OccurredEventArgs)
            Process.Start(executablePath)
        End Sub

    End Module

End Namespace

File: 2000 98 & ME/QuickStart/howto/samples/Services/Timers/scheduler/scheduler.cs
using System;
using System.Diagnostics;
using System.Threading;
using System.Timers;
using System.Timers.Design;
using System.WinForms;
	
public class Scheduler
{
    static string executablePath;
    
    public static void Main(String[] args) 
    {
        string sampleName = Environment.GetCommandLineArgs()[0];

        if(args.Length != 1)
        {
            Console.WriteLine("Usage: " + sampleName +" <executable>");
            return;
        }
        
        executablePath = args[0];
        
        Schedule aSchedule = new Schedule(); 
        RecurrencePatternsDialog dlg = new RecurrencePatternsDialog();
        dlg.RecurrencePatterns = aSchedule.RecurrencePatterns;
        dlg.ShowDialog();
        
        aSchedule.RecurrencePatterns.All = dlg.RecurrencePatterns.All; 
        aSchedule.EventOccurred += new OccurredEventHandler(OnScheduledEvent);
        aSchedule.Enabled = true;
                
        Console.WriteLine("Press \'q\' to quit the sample");	
        while(Console.Read()!='q')
        {
            Thread.Sleep(500);	
        }
    }

	public static void OnScheduledEvent(Object source, OccurredEventArgs e)
	{
        Process.Start(executablePath);		
    }
}